packages/help-center/src/hooks/use-current-page-search-mapping.tsx

```diff
@@ -1,7 +1,45 @@
+import { store as blocksStore } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
 import currentPageSearchMapping from '../route-to-query-mapping.json';
 
+interface CoreBlockEditor {
+	getSelectedBlock: () => object;
+}
+
+interface BlockStore {
+	getBlockTypes: () => [ value: { name: string } ];
+}
+
 export function useCurrentPageSearchMapping( currentRoute: string | undefined ) {
 	let searchQuery = '';
+
+	const selectedBlock = useSelect(
+		( select: ( store: string ) => CoreBlockEditor ) =>
+			select( 'core/block-editor' )?.getSelectedBlock(),
+		[]
+	);
+
+	const blockTypes = useSelect(
+		( select: ( store: string ) => BlockStore ) => select( blocksStore )?.getBlockTypes(),
+		[]
+	);
+
+	// Set search query based on selected block in editor
+	if ( selectedBlock && blockTypes ) {
+		blockTypes.forEach( ( blockType ) => {
+			const blockName = blockType.name;
+			if ( ( selectedBlock as { name: string } ).name === blockName ) {
+				searchQuery = blockName;
+				return;
+			}
+		} );
```

Comment:
[Proposal]: I think that this would be more clear if we use javascript's [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) function.
What do you think?

```javascript
const searchQuery = blockTypes?.find( blockType => blockType.name === selectedBlock.name)?.name
```